pkg/build/builder/sti.go

```diff
@@ -17,24 +18,73 @@ import (
 	"github.com/openshift/origin/pkg/build/builder/cmd/dockercfg"
 )
 
+// internal interface to decouple S2I-specific code from Origin builder code
+type stiBuilderFactory interface {
+	// retrieve S2I Builder based on S2I configuration
+	GetStrategy(config *stiapi.Config) (build.Builder, error)
+}
+
+// interval interface to decouple S2I-specific code from Origin builder code
+type stiConfigValidator interface {
+	ValidateConfig(config *stiapi.Config) []validation.ValidationError
+}
+
+// default implementation of stiBuilderFactory
+type runtimeBuilderFactory struct{}
+
+// default implementation of stiConfigValidator
+type runtimeConfigValidator struct{}
+
+// default implementation of stiBuildFactory.GetStrategy method. Just delegates to S2I-specific code
```

Comment:
GetStrategy is the default implementation... (you have to prefix with function name to have valid godoc)